docs/pages/machine-id/architecture.mdx

```diff
@@ -0,0 +1,148 @@
+differences.
+
+Static Token:
+
+- The name of the token is the only value needed to join. This value is a
+  secret and must be handled securely.
+- Once used, the token resource self-destructs. This means it can only be used 
+  to join a single bot to a Teleport cluster.
+- The certificates exchanged for the token are 
+  [renewable](#renewable-certificates).
+
+
+Dynamic join tokens (e.g AWS IAM):
+
+- These tokens rely on some kind of external system that allows the bot to
+  prove it is allowed to join the cluster. The name of the token identifies the
+  Token resource in Teleport that contains the configuration.
+- The token can be used to join as many bots as you want.
+- The certificates exchanged for the token are not renewable. When the bot wants
+  to renew its certificates, it simply repeats the original join steps.
+
+Where possible, you should prefer to use a dynamic join token over a static
+token as this eliminates the need to handle a secret.
+
+## Renewable certificates
+
+Human Teleport users are not able to renew their credentials without
+re-authenticating. As Machine ID must work without manual human intervention,
+bots that have joined using static tokens are granted credentials that are
+marked as renewable. This means that the bot can use these credentials to
+fetch a new set of credentials with an expiry date further in the future.
+
+In order to mitigate the risk of bot user credentials being stolen, and then
+continually renewed by a malicious actor, renewable bot user certificates
+include a **generation counter**.
+
+The generation counter is stored against the user in the database and within the
+certificate. This counter is incremented each time the user renews their 
+certificate. When a bot attempts to renew, the Auth Server ensures that the 
+value within the certificate and in the database match. If they do not match,
+then the bot user is automatically locked. This means that if certificates are
+stolen, and attempted to be renewed whilst the bot is still running, the next
+renewal will render them useless.
+
+## File permissions
+
+There are two types of folder in use by `tbot`:
+
+- The bot's own files: these store sensitive credentials belonging to the bot
+  user itself. This is by default stored at `/var/lib/teleport/bot/`.
+- Directory destinations: when a directory destination is configured, the bot 
+  outputs the role impersonated credentials as files in the specified directory.
+
+It is important that we ensure that these files can only be accessed by the
+fewest number of users on the system as is necessary.
+
+In the case of the bot's own files, this should only be the user that the bot
+runs as. Running `tbot init` as shown in the documentation should create and
+correctly configure the permissions for this.
+
+In the case of directory destinations, the user the bot runs as requires read
+and write permissions, and users that will need to use the credentials output
```

Comment:
nit: I think there's at least one too many senses of "user" at play here. Maybe "processes" or "readers" or something?

Reply:
Hmm, I agree, although "user" is definitely the most technically correct term, but "process" should be clear enough.

Reply:
f4ba44362